To: vim_dev@googlegroups.com
Subject: Patch 7.3.444
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.444
Problem:    ":all!" and ":sall!" give error E477, even though the
            documentation says these are valid commands.
Solution:   Support the exclamation mark. (Hirohito Higashi)
Files:      src/ex_cmds.h, src/testdir/test31.in, src/testdir/test31.ok


*** ../vim-7.3.443/src/ex_cmds.h	2011-06-20 05:02:53.000000000 +0200
--- src/ex_cmds.h	2012-02-12 23:53:31.000000000 +0100
***************
*** 102,108 ****
  EX(CMD_aboveleft,	"aboveleft",	ex_wrongmodifier,
  			NEEDARG|EXTRA|NOTRLCOM),
  EX(CMD_all,		"all",		ex_all,
! 			RANGE|NOTADR|COUNT|TRLBAR),
  EX(CMD_amenu,		"amenu",	ex_menu,
  			RANGE|NOTADR|ZEROR|EXTRA|TRLBAR|NOTRLCOM|USECTRLV|CMDWIN),
  EX(CMD_anoremenu,	"anoremenu",	ex_menu,
--- 102,108 ----
  EX(CMD_aboveleft,	"aboveleft",	ex_wrongmodifier,
  			NEEDARG|EXTRA|NOTRLCOM),
  EX(CMD_all,		"all",		ex_all,
! 			BANG|RANGE|NOTADR|COUNT|TRLBAR),
  EX(CMD_amenu,		"amenu",	ex_menu,
  			RANGE|NOTADR|ZEROR|EXTRA|TRLBAR|NOTRLCOM|USECTRLV|CMDWIN),
  EX(CMD_anoremenu,	"anoremenu",	ex_menu,
***************
*** 798,804 ****
  EX(CMD_sargument,	"sargument",	ex_argument,
  			BANG|RANGE|NOTADR|COUNT|EXTRA|EDITCMD|ARGOPT|TRLBAR),
  EX(CMD_sall,		"sall",		ex_all,
! 			RANGE|NOTADR|COUNT|TRLBAR),
  EX(CMD_sandbox,		"sandbox",	ex_wrongmodifier,
  			NEEDARG|EXTRA|NOTRLCOM),
  EX(CMD_saveas,		"saveas",	ex_write,
--- 798,804 ----
  EX(CMD_sargument,	"sargument",	ex_argument,
  			BANG|RANGE|NOTADR|COUNT|EXTRA|EDITCMD|ARGOPT|TRLBAR),
  EX(CMD_sall,		"sall",		ex_all,
! 			BANG|RANGE|NOTADR|COUNT|TRLBAR),
  EX(CMD_sandbox,		"sandbox",	ex_wrongmodifier,
  			NEEDARG|EXTRA|NOTRLCOM),
  EX(CMD_saveas,		"saveas",	ex_write,
*** ../vim-7.3.443/src/testdir/test31.in	2010-08-15 21:57:29.000000000 +0200
--- src/testdir/test31.in	2012-02-12 23:57:42.000000000 +0100
***************
*** 63,68 ****
--- 63,74 ----
  :" write "testtext 1"
  A 3:close!
  :w >>test.out
+ :set nohidden
+ :" test ":all!" hides changed buffer; write "testtext 2 2 2"
+ :sp Xtest4
+ GA 4:all!
+ :1wincmd w
+ :w >>test.out
  :qa!
  ENDTEST
  
*** ../vim-7.3.443/src/testdir/test31.ok	2010-08-15 21:57:29.000000000 +0200
--- src/testdir/test31.ok	2012-02-13 00:00:13.000000000 +0100
***************
*** 9,11 ****
--- 9,12 ----
  testtext 1
  testtext 3 3 3
  testtext 1
+ testtext 2 2 2
*** ../vim-7.3.443/src/version.c	2012-02-12 23:23:25.000000000 +0100
--- src/version.c	2012-02-12 23:52:47.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     444,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
50. The last girl you picked up was only a jpeg.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
